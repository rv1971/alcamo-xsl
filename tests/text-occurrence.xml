<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/text.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
      </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="concat(@minOccurs, ', ', @maxOccurs)"/>
        <xsl:call-template select="." name="a:occurrence"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item/>
    <a:item maxOccurs="1"/>
    <a:item minOccurs="0" maxOccurs="1"/>
    <a:item minOccurs="0" maxOccurs="2"/>
    <a:item minOccurs="0" maxOccurs="3"/>
    <a:item minOccurs="1"/>
    <a:item minOccurs="001" maxOccurs=" 01"/>
    <a:item minOccurs="1" maxOccurs="2"/>
    <a:item minOccurs="1" maxOccurs="3"/>
    <a:item minOccurs="1" maxOccurs="10"/>
    <a:item minOccurs="1" maxOccurs="11"/>
    <a:item minOccurs="2" maxOccurs="2"/>
    <a:item minOccurs="2" maxOccurs="3"/>
    <a:item minOccurs="2" maxOccurs="4"/>
    <a:item minOccurs="2" maxOccurs="11"/>
    <a:item minOccurs="11" maxOccurs="11"/>
    <a:item minOccurs="11" maxOccurs="12"/>
  </a:data>
</a:doc>
