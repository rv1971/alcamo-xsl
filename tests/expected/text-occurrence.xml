<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>, : </a:item>
  <a:item>, 1: </a:item>
  <a:item>0, 1: optional</a:item>
  <a:item>0, 2: zero to two</a:item>
  <a:item>0, 3: zero to three</a:item>
  <a:item>1, : </a:item>
  <a:item>001,  01: </a:item>
  <a:item>1, 2: one to two</a:item>
  <a:item>1, 3: one to three</a:item>
  <a:item>1, 10: one to ten</a:item>
  <a:item>1, 11: one to 11</a:item>
  <a:item>2, 2: exactly two</a:item>
  <a:item>2, 3: two to three</a:item>
  <a:item>2, 4: two to four</a:item>
  <a:item>2, 11: two to 11</a:item>
  <a:item>11, 11: exactly 11</a:item>
  <a:item>11, 12: 11 to 12</a:item>
</a:result>
